### PR TITLE
Run blazor create/build/publish as part of template tests

### PR DESCRIPTION
**Pull request by HaoK**

### CODE REVIEW

File: src/ProjectTemplates/test/BlazorTemplateTest.cs [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+                {
+                    targetProject = GetSubProject(project, "Server", $"{project.ProjectName}.Server");
+                }
+
+                var publishResult = await targetProject.RunDotNetPublishAsync(noRestore: false);
+                Assert.True(0 == publishResult.ExitCode, ErrorMessages.GetFailedProcessMessage("publish", targetProject, publishResult));
+
+                // Run dotnet build after publish. The reason is that one uses Config = Debug and the other uses Config = Release
+                // The output from publish will go into bin/Release/netcoreappX.Y/publish and won't be affected by calling build
+                // later, while the opposite is not true.
```
Comment (pranavkm) [inline src/ProjectTemplates/test/BlazorTemplateTest.cs:71]: This is patently false btw.

**Comment by HaoK** [comment]
Create/build/publish tests are added back to ProjectTemplates, these tests are currently not running on helix again due to the last runtime dependency update, so you can see the new tests running in the Wdinws_testTempates_Results: 

image

File: src/ProjectTemplates/test/BlazorWasmTemplateTest.cs [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+            var serverProject = GetSubProject(project, "Server", $"{project.ProjectName}.Server");
+
+            var serverProjectFileContents = ReadFile(serverProject.TemplateOutputDir, $"{serverProject.ProjectName}.csproj");
+            if (!useLocalDb)
+            {
+                Assert.Contains(".db", serverProjectFileContents);
+            }
+
+            var appSettings = ReadFile(serverProject.TemplateOutputDir, "appsettings.json");
+            var element = JsonSerializer.Deserialize<JsonElement>(appSettings);
```
Comment (pranavkm) [inline src/ProjectTemplates/test/BlazorWasmTemplateTest.cs:105]: Any chance you could only use STJ? Most of the things this is doing should be achievable using a JsonDocument or using an actual type.

**Review by pranavkm — APPROVED** [review]

File: src/ProjectTemplates/test/BlazorWasmTemplateTest.cs (hunk 2 above)
Comment (HaoK) [reply src/ProjectTemplates/test/BlazorWasmTemplateTest.cs:105]: I just lifted this from the existing template tests, do you want me to file a cleanup issue to switching both of these?

File: src/ProjectTemplates/test/BlazorTemplateTest.cs (hunk 1 above)
Comment (HaoK) [reply src/ProjectTemplates/test/BlazorTemplateTest.cs:71]: Remove or is there updated text?

Comment (pranavkm) [reply src/ProjectTemplates/test/BlazorTemplateTest.cs:71]: Remove, you should be able to build and publish in any Configuration.

Comment (HaoK) [reply src/ProjectTemplates/test/BlazorTemplateTest.cs]: Cool, I deleted it from the current blazor template tests in quarantine as well, so we no longer build after publish

**Pull request merged**